app/containers/Menu.js

```diff
@@ -0,0 +1,254 @@
+var React = require('react');
+var styles = require('../styles');
+var ReactRouter = require('react-router');
+var Link = ReactRouter.Link;
+var Editor = require('./Editor');
+
+var Menu = React.createClass({
+
+  getInitialState: function () {
+
+    return {
+      lastSort: 'Alphabetical',
+      sort: "Popularity",
+      popularButtonDisabled: true,
+      alphabeticalButtonDisabled: false,
+      editDrinkName: '',
+      editDrinkIngredients: '',
+      drinkName: '',
+      newDrinkIngredients: '',
+      drinkList: JSON.parse(localStorage.getItem('drinkList')),
+      isEditing: 'Edit',
+
+    };
+  },
+
+ handleDrinkNameChange: function (e) {
+    this.setState({
+      drinkName: e.target.value,
+    }, function() {
+      console.log(this.state.drinkName);
+    });
+  },
+
+  handleNewIngredientsChange: function (e) {
+    this.setState({
+      newdrinkingredients: e.target.value,
+    }, function() {
+      console.log(this.state.newdrinkingredients);
+    });
+  },
+
+  handleSubmit: function (e) {
+    if(this.state.drinkName !== '') {
+    var urlName = this.state.drinkName.replace(/\s/g, '+');
+    e.preventDefault();
+    var drinkName = this.state.drinkName;
+    this.state.drinkList.push({name: this.state.drinkName, ingredients: this.state.newdrinkingredients, rating: 5, drinkNumber: this.state.drinkList.length, url: 'http://google.com/#q=' + urlName + '+cocktail'});
+    this.setState({
+      drinklist: this.state.drinkList,
+      drinkName: '',
+      newdrinkingredients: '',
+    });
+  } else {
+    this.setState({
+      drinkName: '',
+      newdrinkingredients: '',
+    });
+  }
+
+  },
+
+  handlePopSort: function () {
+    this.setState({
+      drinkList: this.popularSort(),
+    });
+  },
+
+  handleAlphSort: function () {
+    this.setState({
+      drinkList: this.alphabeticalSort(),
+    });
+  },
+
+  removeDrink: function (name) {
+    var index = this.state.drinkList.indexOf(name);
+    this.state.drinkList.splice(index, 1);
+    this.setState({
+      drinkList: this.state.drinkList,
+    });
+  },
+
+  editName: function (e) {
+    this.setState({
+      drinkList: e.target.value,
+    });
+    console.log(this.state.editDrinkName);
+    /*var index = this.state.drinkList.indexOf(name);
+    this.state.drinkList[index].name = this.state.editDrinkName;
+    this.setState({
+      drinkList: this.state.drinkList,
+    });*/
+
+  },
+
+  editDrinkList: function (e) {
+    console.log(this.state.drinkList.indexOf(e));
+  },
+
+  /*editIngredients: function (name) {
+    console.log(this.state.editIngredients);
+    var index = this.state.drinkList.indexOf(name);
+    this.state.drinkList[index].ingredients = this.state.EditDrinkIngredients;
+    this.setState({
+      drinkList: this.state.drinkList,
+    });
+  },*/
+
+  drinkListReturn: function () {
+  if (this.state.isEditing === 'Save') {
```

Comment:
thankfully, there are type checkers that make javascript more static (type of variable is known at compile time) checkout [Flow from Facebook](https://flowtype.org/)